lib/streamlit/runtime/state/query_params_proxy.py

```diff
@@ -0,0 +1,86 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Dict, Iterator, List, Union
+
+from streamlit.runtime.state.query_params import (
+    QueryParams,
+    _missing_key_error_message_query_params,
+)
+from streamlit.runtime.state.session_state_proxy import get_session_state
+from streamlit.type_util import Key
+
+
+def get_query_params() -> QueryParams:
+    from streamlit.runtime.scriptrunner import get_script_run_ctx
+
+    ctx = get_script_run_ctx()
+
+    if ctx is None:
+        return QueryParams()
+    return get_session_state()._state._query_params
+
+
+class QueryParamsProxy:
+    """A stateless singleton that proxies `st.query_params` interactions
+    to the current script thread's QueryParams instance.
+    """
+
+    def __iter__(self) -> Iterator[Any]:
+        return iter(get_query_params())
```

Comment:
We know that this is an `Iterator[Union[str, List[str]]]`, right?

Reply:
I think yes but I got a type error as we are overriding. I think we should probably just leave it but happy to change.
```
lib/streamlit/runtime/state/query_params_proxy.py:36:5: error: Return type "Iterator[Union[str, List[str]]]" of "__iter__" incompatible with
return type "Iterator[str]" in supertype "Iterable"  [override]
        def __iter__(self) -> Iterator[Union[str, List[str]]]:
```

Reply:
I think the error comes from `__getitem__` only returning type `str`, which confuses `mypy` since the `__getitem__` function's return type seems to conflict. Worth overriding rather than returning the wrong type